src/lib.rs

```diff
@@ -44,13 +57,20 @@ pub struct CanisterState {
 }
 
 #[ic_cdk_macros::update]
-/// Deposits funds for the specified participant into the specified channel.
-/// Please do NOT over-fund or fund channels that are already fully funded, as
-/// this can lead to a permanent LOSS OF FUNDS.
-fn deposit(funding: Funding, amount: Amount) -> Option<Error> {
-	STATE
-		.with(|s| s.borrow_mut().deposit(funding, amount))
-		.err()
+/// The user needs to call this with his transaction.
+async fn transaction_notification(block_height: u64) {
```

Comment:
```suggestion
async fn process_funding_tx(block_height: u64) {
```
?

Reply:
Combine with `deposit`?

Reply:
Currently I need it separately for easier testing.